lib/Graph/Graph.cpp

```diff
@@ -38,14 +38,18 @@ using llvm::isa;
 
 namespace {
 /// A helper function to log the deletion of constant/placeholder \p s of a
-/// module.
-void loggingStorageDeletionInUserFunc(Storage *s) {
-  std::unordered_set<Function *> userFuncs;
-  for (auto &use : s->getUsers()) {
-    userFuncs.insert(use.getUser()->getParent());
+/// module into the log context of given functions \p functions.
+void loggingStorageDeletion(std::list<Function *> functions, Storage *s) {
+  for (auto *F : functions) {
+    F->getLogContext().logNodeDeletion(*s, /* logIntoModule */ true);
```

Comment:
Won't this log the deletion into the module log once for every function? Is that intentional?

Reply:
thx for reminding. This is not intentional. I just fixed that logging into module will only happen once.